Add release notes for 6.1.0 (#2762)
sensu/sensu-docs: content/sensu-go/6.1/release-notes.md

---

## 6.1.0 release notes

**October 5, 2020** &mdash; The latest release of Sensu Go, version 6.1.0, is now available for download.

**PLACEHOLDER FOR RELEASE SYNOPSIS**

See the [upgrade guide][1] to upgrade Sensu to version 6.1.0.

**NEW FEATURES:**

- ([Commercial feature][172]) Added support for custom secrets engine paths in [Vault secrets][175].
- ([Commercial feature][172]) In the web UI, added [new search functionality][178], with improved syntax and suggestions.

> - ([Commercial feature][172]) Added [`strict` field to PostgresConfig][179] to help debug incorrect configurations and database permissions.

Author: ```suggestion
- ([Commercial feature][172]) Added [`strict` attribute][179] to the PostgresConfig type to help debug incorrect configurations and database permissions.
- ([Commercial feature][172]) Added [`batch_buffer`, `batch_size`, and `batch_buffer` attributes][181] to the PostgresConfig type so operators can optimize PostgreSQL latency and throughput.
```

- ([Commercial feature][172]) Added TLS configuration to the cluster resource so you can specify additional CA certificates and insecure mode.
- ([Commercial feature][172]) Added a `types` query parameter for listing [authentication providers][176] and [secrets providers][177] via the API.
- ([Commercial feature][172]) The Alpine-based Docker image now has multi-architecture support with support for the linux/386, linux/amd64, linux/arm64, linux/arm/v6, linux/arm/v7, linux/ppc64le, and linux/s390x platforms.
- The backend flag [`--api-request-limit`][173] is now available to configure the maximum API request body size (in bytes).
- In the [REST API][174], most configuration resources now support the PATCH method for making updates.

**IMPROVEMENTS:**

> - ([Commercial feature][172]) Improved logging for OIDC authentication providers.

Author: ```suggestion
- ([Commercial feature][172]) Improved logging for OIDC authentication providers. Also added [`disable_offline_access` OIDC spec attribute][180], which allows OIDC providers to return a refresh token so users can refresh their access tokens.
```

---

## 6.1.0 release notes

**October 5, 2020** &mdash; The latest release of Sensu Go, version 6.1.0, is now available for download.

**PLACEHOLDER FOR RELEASE SYNOPSIS**

See the [upgrade guide][1] to upgrade Sensu to version 6.1.0.

**NEW FEATURES:**

- ([Commercial feature][172]) Added support for custom secrets engine paths in [Vault secrets][175].
- ([Commercial feature][172]) In the web UI, added [new search functionality][178], with improved syntax and suggestions.
- ([Commercial feature][172]) Added [`strict` attribute][179] to the PostgresConfig type to help debug incorrect configurations and database permissions.

> - ([Commercial feature][172]) Added [`batch_buffer`, `batch_size`, and `batch_buffer` attributes][181] to the PostgresConfig type so operators can optimize PostgreSQL latency and throughput.

Author: ```suggestion
- ([Commercial feature][172]) Added [`batch_buffer`, `batch_size`, and `batch_workers` attributes][181] to the PostgresConfig type so operators can optimize PostgreSQL latency and throughput.
```

- ([Commercial feature][172]) Added TLS configuration to the cluster resource so you can specify additional CA certificates and insecure mode.
- ([Commercial feature][172]) Added a `types` query parameter for listing [authentication providers][176] and [secrets providers][177] via the API.
- ([Commercial feature][172]) The Alpine-based Docker image now has multi-architecture support with support for the linux/386, linux/amd64, linux/arm64, linux/arm/v6, linux/arm/v7, linux/ppc64le, and linux/s390x platforms.
- The backend flag [`--api-request-limit`][173] is now available to configure the maximum API request body size (in bytes).
- In the [REST API][174], most configuration resources now support the PATCH method for making updates.

**IMPROVEMENTS:**

- ([Commercial feature][172]) Improved logging for OIDC authentication providers. Also added [`disable_offline_access` OIDC spec attribute][180], which allows OIDC providers to return a refresh token so users can refresh their access tokens.
- ([Commercial feature][172]) Added indexed field and label selectors to the PostgreSQL event store to improve performance for PostgreSQL event store queries with field and label selectors.

> - Added Prometheus transformer for extracting metrics from check output using the Prometheus Exposition Text Format.

Author: ```suggestion
- Added Prometheus transformer for extracting metrics from check output using the Prometheus Exposition Text Format.
- Added the [`output_metric_tags` attribute][182] for checks so you can apply custom tags to enrich metric points produced by check output metric extraction.
```

- ([Commercial feature][172]) Added [`strict` attribute][179] to the PostgresConfig type to help debug incorrect configurations and database permissions.
- ([Commercial feature][172]) Added [`batch_buffer`, `batch_size`, and `batch_workers` attributes][181] to the PostgresConfig type so operators can optimize PostgreSQL latency and throughput.
- ([Commercial feature][172]) Added TLS configuration to the cluster resource so you can specify additional CA certificates and insecure mode.
- ([Commercial feature][172]) Added a `types` query parameter for listing [authentication providers][176] and [secrets providers][177] via the API.
- ([Commercial feature][172]) The Alpine-based Docker image now has multi-architecture support with support for the linux/386, linux/amd64, linux/arm64, linux/arm/v6, linux/arm/v7, linux/ppc64le, and linux/s390x platforms.
- The backend flag [`--api-request-limit`][173] is now available to configure the maximum API request body size (in bytes).
- In the [REST API][174], most configuration resources now support the PATCH method for making updates.

**IMPROVEMENTS:**

> - ([Commercial feature][172]) Improved logging for OIDC authentication providers. Also added [`disable_offline_access` OIDC spec attribute][180], which allows OIDC providers to return a refresh token so users can refresh their access tokens.

Reviewer: I would maybe change the description for `disable_offline_access`, to mention that it's basically a workaround for authorization servers that do not support the `offline_access` scope.

> - ([Commercial feature][172]) Improved logging for OIDC authentication providers. Also added [`disable_offline_access` OIDC spec attribute][180], which allows OIDC providers to return a refresh token so users can refresh their access tokens.

Author: ```suggestion
- ([Commercial feature][172]) Improved logging for OIDC authentication providers. Also added [`disable_offline_access` OIDC spec attribute][180], which provides a workaround for authorization servers that do not support the `offline_access` scope.
```

- A warning is now logged when you request a dynamic runtime asset that does not exist.
- The trusted CA file is now used for agent, backend, and sensuctl asset retrieval.
- Per-entity subscriptions (such as `entity:entityName`) are always available for agent entities, even you remove subscriptions via the entities API.

**FIXES:**

- ([Commercial feature][172]) Fixed a bug in `sensuctl dump` that allowed polymorphic resources (e.g., secrets providers and authentication providers) to dump other providers of the same type.
- ([Commercial feature][172]) Check output is no longer truncated in the event log file when the max output size is set and the PostgreSQL event store is enabled.
- ([Commercial feature][172]) Sensuctl prune now handles multi-file/multi-url input correctly.
- ([Commercial feature][172]) Fixed a bug where PostgreSQL errors could cause the backend to panic.
- ([Commercial feature][172]) Fixed a bug where PostgreSQL would refuse to store event with a negative check status.
- The backend will no longer start if the web UI TLS configuration is not fully specified.
- The agent entity is now included in data passed to the STDIN for the command process.

> - Proxy entities used in round robin check requests are no longer stale.

Reviewer 2: I feel like this needs a reworking -- it sounds weird to me as written.

Looks like it refers to https://github.com/sensu/sensu-go/pull/4008, where I think Nikki's first description better captures the change:

"Ensures that the interval/cron round robin schedulers do not use stale proxy entities when scheduling check requests."

Maybe something like

```suggestion
Improved check scheduling to prevent stale proxy entity data when using cron or round robin schedulers.
```

> Improved check scheduling to prevent stale proxy entity data when using cron or round robin schedulers.

Author: ```suggestion
- Improved check scheduling to prevent stale proxy entity data when using cron or round robin schedulers.
```

---

## 6.1.0 release notes

**October 5, 2020** &mdash; The latest release of Sensu Go, version 6.1.0, is now available for download.

> **PLACEHOLDER FOR RELEASE SYNOPSIS**

Author: ```suggestion
The latest release of Sensu Go, version 6.1.0, is now available for download! This release delivers significant performance and stability gains, feature enhancements, and several bug fixes. The web UI is now much snappier, and its search is redesigned with an improved syntax and suggestions! Monitor even more services and infrastructure when using the PostgreSQL store: batched Sensu event writes and improved indexing allows a single Sensu Go deployment to process and query more data than ever before. If you’re using Prometheus client libraries to instrument your applications, the Sensu Go agent can now scrape and enrich those metrics! And if you’re collecting metrics in other formats like Nagios PerfData, you can use the new output metric tags feature to enrich those metrics too! The sensuctl prune command also received some love, and it now loads and prunes configuration resources from multiple files!
```

> The latest release of Sensu Go, version 6.1.0, is now available for download! This release delivers significant performance and stability gains, feature enhancements, and several bug fixes. The web UI is now much snappier, and its search is redesigned with an improved syntax and suggestions! Monitor even more services and infrastructure when using the PostgreSQL store: batched Sensu event writes and improved indexing allows a single Sensu Go deployment to process and query more data than ever before. If you’re using Prometheus client libraries to instrument your applications, the Sensu Go agent can now scrape and enrich those metrics! And if you’re collecting metrics in other formats like Nagios PerfData, you can use the new output metric tags feature to enrich those metrics too! The sensuctl prune command also received some love, and it now loads and prunes configuration resources from multiple files!

Author: ```suggestion
This release delivers significant performance and stability gains, feature enhancements, and several bug fixes. The web UI is now much snappier, and its search is redesigned with an improved syntax and suggestions! Monitor even more services and infrastructure when using the PostgreSQL store: batched Sensu event writes and improved indexing allows a single Sensu Go deployment to process and query more data than ever before. If you’re using Prometheus client libraries to instrument your applications, the Sensu Go agent can now scrape and enrich those metrics! And if you’re collecting metrics in other formats like Nagios PerfData, you can use the new output metric tags feature to enrich those metrics too! The sensuctl prune command also received some love, and it now loads and prunes configuration resources from multiple files!
```

[169]: /sensu-go/6.0/observability-pipeline/observe-filter/filters/#build-event-filter-expressions-with-javascript-execution-functions
[170]: /sensu-go/6.0/operations/maintain-sensu/upgrade/#upgrade-to-sensu-go-60-from-a-5x-deployment
[171]: /sensu-go/6.0/observability-pipeline/observe-entities/entities/#create-and-manage-agent-entities
[172]: /sensu-go/6.1/commercial/
[173]: /sensu-go/6.1/observability-pipeline/observe-schedule/backend/#api-request-limit
[174]: /sensu-go/6.1/api/
[175]: /sensu-go/6.1/operations/manage-secrets/secrets-management/#use-hashicorp-vault-for-secrets-management
[176]: /sensu-go/6.1/api/authproviders/#authproviders-get-specification
[177]: /sensu-go/6.1/api/secrets/#providers-get-specification
[178]: /sensu-go/6.1/web-ui/search/

> [179]: /sensu-go/6.1/operations/deploy-sensu/datastore/#strict

Author: ```suggestion
[179]: /sensu-go/6.1/operations/deploy-sensu/datastore/#strict
[180]: /sensu-go/6.1/operations/control-access/auth/#oidc-spec-attributes
```

[169]: /sensu-go/6.0/observability-pipeline/observe-filter/filters/#build-event-filter-expressions-with-javascript-execution-functions
[170]: /sensu-go/6.0/operations/maintain-sensu/upgrade/#upgrade-to-sensu-go-60-from-a-5x-deployment
[171]: /sensu-go/6.0/observability-pipeline/observe-entities/entities/#create-and-manage-agent-entities
[172]: /sensu-go/6.1/commercial/
[173]: /sensu-go/6.1/observability-pipeline/observe-schedule/backend/#api-request-limit
[174]: /sensu-go/6.1/api/
[175]: /sensu-go/6.1/operations/manage-secrets/secrets-management/#use-hashicorp-vault-for-secrets-management
[176]: /sensu-go/6.1/api/authproviders/#authproviders-get-specification
[177]: /sensu-go/6.1/api/secrets/#providers-get-specification
[178]: /sensu-go/6.1/web-ui/search/
[179]: /sensu-go/6.1/operations/deploy-sensu/datastore/#strict

> [180]: /sensu-go/6.1/operations/control-access/auth/#oidc-spec-attributes

Author: ```suggestion
[180]: /sensu-go/6.1/operations/control-access/auth/#oidc-spec-attributes
[181]: /sensu-go/latest/operations/deploy-sensu/datastore/#spec-attributes
```

[169]: /sensu-go/6.0/observability-pipeline/observe-filter/filters/#build-event-filter-expressions-with-javascript-execution-functions
[170]: /sensu-go/6.0/operations/maintain-sensu/upgrade/#upgrade-to-sensu-go-60-from-a-5x-deployment
[171]: /sensu-go/6.0/observability-pipeline/observe-entities/entities/#create-and-manage-agent-entities
[172]: /sensu-go/6.1/commercial/
[173]: /sensu-go/6.1/observability-pipeline/observe-schedule/backend/#api-request-limit
[174]: /sensu-go/6.1/api/
[175]: /sensu-go/6.1/operations/manage-secrets/secrets-management/#use-hashicorp-vault-for-secrets-management
[176]: /sensu-go/6.1/api/authproviders/#authproviders-get-specification
[177]: /sensu-go/6.1/api/secrets/#providers-get-specification
[178]: /sensu-go/6.1/web-ui/search/
[179]: /sensu-go/6.1/operations/deploy-sensu/datastore/#strict
[180]: /sensu-go/6.1/operations/control-access/auth/#oidc-spec-attributes

> [181]: /sensu-go/latest/operations/deploy-sensu/datastore/#spec-attributes

Author: ```suggestion
[181]: /sensu-go/latest/operations/deploy-sensu/datastore/#spec-attributes
[182]: /sensu-go/6.1/observability-pipeline/observe-schedule/checks#output-metric-tags
```
